checker/src/org/checkerframework/checker/signedness/SignednessVisitor.java

```diff
@@ -91,6 +106,73 @@ private boolean maskIgnoresMSB(Kind maskKind, LiteralTree numBitsLit, LiteralTre
     }
 
     /**
+     * Given a casted right shift of the form {@code (type) (baseExpr >> numBits)} or {@code (type)
+     * (baseExpr >>> numBits)}, return true iff the expression's value is the same regardless of the
+     * type of right shift (signed or unsigned). This is true if the cast ignores the numBits most
+     * significant bits of the shift result -- that is, if the cast ignores all the new bits that
+     * the right shift introduced on the left.
+     *
+     * <p>For example, the function returns true for
+     *
+     * <pre>{@code (short) (myInt >> 16)}</pre>
+     *
+     * and for
+     *
+     * <pre>{@code (short) (myInt >>> 16)}</pre>
+     *
+     * because these two expressions are guaranteed to have the same result.
+     *
+     * @param shiftTypeKind the kind of the type of the shift literal (BYTE, CHAR, SHORT, INT, or
+     *     LONG)
+     * @param castTypeKind the kind of the cast target type (BYTE, CHAR, SHORT, INT, or LONG)
+     * @param numBitsLit the LiteralTree whose value is numBits
+     * @return true iff introduced bits are discarded
+     */
+    private boolean castIgnoresMSB(
+            TypeKind shiftTypeKind, TypeKind castTypeKind, LiteralTree numBitsLit) {
```

Comment:
`numBits` is a confusing name, because in the body of the routine there are a lot of variables that count a number of bits.  When you name a variable, name it according to its *semantics*, not its *type*.  (The type here is the number of bits, which is more informative than naming it `i` for `int` but still doesn't get at the semantics.)  I would call it `shiftAmount` or something like that.